src/main/java/seedu/address/model/task/DeadlineTask.java

```diff
@@ -3,28 +3,49 @@
 import java.time.LocalDateTime;
 import java.util.Objects;
 
+import java.util.function.Predicate;
+
 import seedu.address.commons.exceptions.IllegalValueException;
 
 public class DeadlineTask extends Task {
 
-    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s]";
+    private static final String FMT_STRING = "DeadlineTask[name=%s, due=%s, finished=%s]";
 
     private final LocalDateTime due;
 
-    public DeadlineTask(Name name, LocalDateTime due) {
+    private boolean finished;
+
+    public static Predicate<? super DeadlineTask> isNotFinishedDeadline() {
+        return p -> !p.isFinished();
+    }
+
+    public DeadlineTask(Name name, LocalDateTime due, boolean finished) {
         super(name);
         assert due != null;
         this.due = due;
+        this.finished = finished;
+    }
+
+    public DeadlineTask(Name name, LocalDateTime due) {
+        this(name, due, false);
     }
 
     public DeadlineTask(String name, LocalDateTime due) throws IllegalValueException {
-        this(new Name(name), due);
+        this(new Name(name), due, false);
+    }
+
+    public DeadlineTask(DeadlineTask oldTask, boolean finishStatus) {
+        this(oldTask.name, oldTask.getDue(), finishStatus);
```

Comment:
Ugh, I don't think these kind of constructors should be introduced, because then for consistency (and consistency is important for good code) we will have to introduce constructors for every combination of fields that we potentially want to edit. That is, we need to implement 7 constructors!

Either just use `new DeadlineTask(oldName, oldDue, newCompleted)`, or use the builder pattern